planner/mapper/mapper.go

```diff
@@ -119,36 +120,91 @@ func resolveOrderDependencies(
 
 	// If there is orderby, and any one of the condition fields that are join fields and have not been
 	// requested, we need to map them here.
+outer:
 	for _, condition := range source.Value().Conditions {
-		if len(condition.Fields) <= 1 {
-			continue
-		}
-
-		joinField := condition.Fields[0]
+		fields := condition.Fields[:] // copy slice
+		for {
+			numFields := len(fields)
+			if numFields == 2 {
+				joinField := condition.Fields[0]
+
+				// ensure the child select is resolved for this order join
+				innserSelect, err := resolveChildOrder(descriptionsRepo, descName, joinField, mapping, existingFields)
+				if err != nil {
+					return err
+				}
 
-		// Check if the join field is already mapped, if not then map it.
-		if isOrderJoinFieldMapped := len(mapping.IndexesByName[joinField]) != 0; !isOrderJoinFieldMapped {
-			index := mapping.GetNextIndex()
-			mapping.Add(index, joinField)
+				// make sure the actual target field inside the join field
+				// is included in the select
+				targetFieldName := condition.Fields[1]
+				targetField := &Field{
+					Index: innserSelect.FirstIndexOfName(targetFieldName),
+					Name:  targetFieldName,
+				}
+				innserSelect.Fields = append(innserSelect.Fields, targetField)
+				continue outer
+			} else if numFields > 2 {
+				joinField := condition.Fields[0]
 
-			// Resolve the inner child fields and get it's mapping.
-			dummyJoinFieldSelect := request.Select{
-				Field: request.Field{
-					Name: joinField,
-				},
-			}
-			innerSelect, err := toSelect(descriptionsRepo, index, &dummyJoinFieldSelect, descName)
-			if err != nil {
-				return err
+				// ensure the child select is resolved for this order join
+				innserSelect, err := resolveChildOrder(descriptionsRepo, descName, joinField, mapping, existingFields)
+				if err != nil {
+					return err
+				}
+				mapping = &innserSelect.DocumentMapping
+				fields = fields[1:] // chop off the front item, and loop again on inner
+			} else { // <= 1
+				// nothing todo, continue the outer for loop
+				continue outer
 			}
-			*existingFields = append(*existingFields, innerSelect)
-			mapping.SetChildAt(index, &innerSelect.DocumentMapping)
 		}
 	}
 
 	return nil
 }
 
+// given a type join field, ensure its mapping exists
+// and add a coorsponding select field(s)
+func resolveChildOrder(
+	descriptionsRepo *DescriptionsRepo,
+	descName string,
+	orderChildField string,
+	mapping *core.DocumentMapping,
+	existingFields *[]Requestable,
+) (*Select, error) {
+	childFieldIndexes := mapping.IndexesByName[orderChildField]
+	// Check if the join field is already mapped, if not then map it.
+	if isOrderJoinFieldMapped := len(childFieldIndexes) != 0; !isOrderJoinFieldMapped {
```

Comment:
suggestion: I'm not sure `isOrderJoinFieldMapped` adds anything positive here, suggest removal and just do `if len(childFieldIndexes) != 0 {`